In [118]:
import pandas as pd
df =pd.read_csv("./kc_house_data.csv")

In [119]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [120]:
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

CLEANING THE DATASET:


Removing duplicate houses and only keeping the latest data and getting rid of older assessments by first transforming the date to datetime, grouping by ids and getting the max date

In [121]:
dup_id=df.id.value_counts().to_frame()
dup_id.loc[dup_id['count']>1]

count
id               
795000620       3
8651510380      2
2568300045      2
9353300600      2
4139480200      2
...           ...
3630120700      2
4031000520      2
4302201085      2
2206700215      2
6021501535      2

[176 rows x 1 columns]

In [122]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [123]:
df['date']=pd.to_datetime(df['date'])
df

id       date     price  bedrooms  bathrooms  sqft_living  \
0      7129300520 2014-10-13  221900.0         3       1.00         1180   
1      6414100192 2014-12-09  538000.0         3       2.25         2570   
2      5631500400 2015-02-25  180000.0         2       1.00          770   
3      2487200875 2014-12-09  604000.0         4       3.00         1960   
4      1954400510 2015-02-18  510000.0         3       2.00         1680   
...           ...        ...       ...       ...        ...          ...   
21608   263000018 2014-05-21  360000.0         3       2.50         1530   
21609  6600060120 2015-02-23  400000.0         4       2.50         2310   
21610  1523300141 2014-06-23  402101.0         2       0.75         1020   
21611   291310100 2015-01-16  400000.0         3       2.50         1600   
21612  1523300157 2014-10-15  325000.0         2       0.75         1020   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
0          5650     1.0           0     0  ...      7        1180   
1          7242     2.0           0     0  ...      7        2170   
2         10000     1.0           0     0  ...      6         770   
3          5000     1.0           0     0  ...      7        1050   
4          8080     1.0           0     0  ...      8        1680   
...         ...     ...         ...   ...  ...    ...         ...   
21608      1131     3.0           0     0  ...      8        1530   
21609      5813     2.0           0     0  ...      8        2310   
21610      1350     2.0           0     0  ...      7        1020   
21611      2388     2.0           0     0  ...      8        1600   
21612      1076     2.0           0     0  ...      7        1020   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0                  0      1955             0    98178  47.5112 -122.257   
1                400      1951          1991    98125  47.7210 -122.319   
2                  0      1933             0    98028  47.7379 -122.233   
3                910      1965             0    98136  47.5208 -122.393   
4                  0      1987             0    98074  47.6168 -122.045   
...              ...       ...           ...      ...      ...      ...   
21608              0      2009             0    98103  47.6993 -122.346   
21609              0      2014             0    98146  47.5107 -122.362   
21610              0      2009             0    98144  47.5944 -122.299   
21611              0      2004             0    98027  47.5345 -122.069   
21612              0      2008             0    98144  47.5941 -122.299   

       sqft_living15  sqft_lot15  
0               1340        5650  
1               1690        7639  
2               2720        8062  
3               1360        5000  
4               1800        7503  
...              ...         ...  
21608           1530        1509  
21609           1830        7200  
21610           1020        2007  
21611           1410        1287  
21612           1020        1357  

[21613 rows x 21 columns]

In [124]:
df=df.groupby('id').max().reset_index()
df

id       date     price  bedrooms  bathrooms  sqft_living  \
0         1000102 2015-04-22  300000.0         6       3.00         2400   
1         1200019 2014-05-08  647500.0         4       1.75         2060   
2         1200021 2014-08-11  400000.0         3       1.00         1460   
3         2800031 2015-04-01  235000.0         3       1.00         1430   
4         3600057 2015-03-19  402500.0         4       2.00         1650   
...           ...        ...       ...       ...        ...          ...   
21431  9842300095 2014-07-25  365000.0         5       2.00         1600   
21432  9842300485 2015-03-11  380000.0         2       1.00         1040   
21433  9842300540 2014-06-24  339000.0         3       1.00         1100   
21434  9895000040 2014-07-03  399900.0         2       1.75         1410   
21435  9900000190 2014-10-30  268950.0         3       1.00         1320   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
0          9373     2.0           0     0  ...      7        2400   
1         26036     1.0           0     0  ...      8        1160   
2         43000     1.0           0     0  ...      7        1460   
3          7599     1.5           0     0  ...      6        1010   
4          3504     1.0           0     0  ...      7         760   
...         ...     ...         ...   ...  ...    ...         ...   
21431      4168     1.5           0     0  ...      7        1600   
21432      7372     1.0           0     0  ...      7         840   
21433      4128     1.0           0     0  ...      7         720   
21434      1005     1.5           0     0  ...      9         900   
21435      8100     1.0           0     0  ...      6         880   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0                  0      1991             0    98002  47.3262 -122.214   
1                900      1947             0    98166  47.4444 -122.351   
2                  0      1952             0    98166  47.4434 -122.347   
3                420      1930             0    98168  47.4783 -122.265   
4                890      1951          2013    98144  47.5803 -122.294   
...              ...       ...           ...      ...      ...      ...   
21431              0      1927             0    98126  47.5297 -122.381   
21432            200      1939             0    98126  47.5285 -122.378   
21433            380      1942             0    98126  47.5296 -122.379   
21434            510      2011             0    98027  47.5446 -122.018   
21435            440      1943             0    98166  47.4697 -122.351   

       sqft_living15  sqft_lot15  
0               2060        7316  
1               2590       21891  
2               2250       20023  
3               1290       10320  
4               1480        3504  
...              ...         ...  
21431           1190        4168  
21432           1930        5150  
21433           1510        4538  
21434           1440        1188  
21435           1000        8100  

[21436 rows x 21 columns]

In [230]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21436 entries, 0 to 21435
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   id             21436 non-null  int64         
 1   date           21436 non-null  datetime64[ns]
 2   price          21436 non-null  float64       
 3   bedrooms       21436 non-null  int64         
 4   bathrooms      21436 non-null  float64       
 5   sqft_living    21436 non-null  int64         
 6   sqft_lot       21436 non-null  int64         
 7   floors         21436 non-null  float64       
 8   waterfront     21436 non-null  int64         
 9   view           21436 non-null  int64         
 10  condition      21436 non-null  int64         
 11  grade          21436 non-null  int64         
 12  sqft_above     21436 non-null  int64         
 13  sqft_basement  21436 non-null  int64         
 14  yr_built       21436 non-null  int64         
 15  yr_renovated   2143

Looking for and removing missing values or outlier mistakes

In [216]:
df.agg(['min','max']).round(1) 

id       date      price  bedrooms  bathrooms  sqft_living  \
min     1000102 2014-05-02    75000.0         0        0.0          290   
max  9900000190 2015-05-27  7700000.0        33        8.0        13540   

     sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
min       520     1.0           0     0  ...      1         290   
max   1651359     3.5           1     4  ...     13        9410   

     sqft_basement  yr_built  yr_renovated  zipcode   lat   long  \
min              0      1900             0    98001  47.2 -122.5   
max           4820      2015          2015    98199  47.8 -121.3   

     sqft_living15  sqft_lot15  
min            399         651  
max           6210      871200  

[2 rows x 21 columns]

In [220]:
df.loc[df['bedrooms']== 33]

id       date     price  bedrooms  bathrooms  sqft_living  \
6021  2402100895 2014-06-25  640000.0        33       1.75         1620   

      sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
6021      6000     1.0           0     0  ...      7        1040   

      sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
6021            580      1947             0    98103  47.6878 -122.331   

      sqft_living15  sqft_lot15  
6021           1330        4700  

[1 rows x 21 columns]

In [229]:
df.drop(index=[6021], axis=0)

id       date     price  bedrooms  bathrooms  sqft_living  \
0         1000102 2015-04-22  300000.0         6       3.00         2400   
1         1200019 2014-05-08  647500.0         4       1.75         2060   
2         1200021 2014-08-11  400000.0         3       1.00         1460   
3         2800031 2015-04-01  235000.0         3       1.00         1430   
4         3600057 2015-03-19  402500.0         4       2.00         1650   
...           ...        ...       ...       ...        ...          ...   
21431  9842300095 2014-07-25  365000.0         5       2.00         1600   
21432  9842300485 2015-03-11  380000.0         2       1.00         1040   
21433  9842300540 2014-06-24  339000.0         3       1.00         1100   
21434  9895000040 2014-07-03  399900.0         2       1.75         1410   
21435  9900000190 2014-10-30  268950.0         3       1.00         1320   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
0          9373     2.0           0     0  ...      7        2400   
1         26036     1.0           0     0  ...      8        1160   
2         43000     1.0           0     0  ...      7        1460   
3          7599     1.5           0     0  ...      6        1010   
4          3504     1.0           0     0  ...      7         760   
...         ...     ...         ...   ...  ...    ...         ...   
21431      4168     1.5           0     0  ...      7        1600   
21432      7372     1.0           0     0  ...      7         840   
21433      4128     1.0           0     0  ...      7         720   
21434      1005     1.5           0     0  ...      9         900   
21435      8100     1.0           0     0  ...      6         880   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0                  0      1991             0    98002  47.3262 -122.214   
1                900      1947             0    98166  47.4444 -122.351   
2                  0      1952             0    98166  47.4434 -122.347   
3                420      1930             0    98168  47.4783 -122.265   
4                890      1951          2013    98144  47.5803 -122.294   
...              ...       ...           ...      ...      ...      ...   
21431              0      1927             0    98126  47.5297 -122.381   
21432            200      1939             0    98126  47.5285 -122.378   
21433            380      1942             0    98126  47.5296 -122.379   
21434            510      2011             0    98027  47.5446 -122.018   
21435            440      1943             0    98166  47.4697 -122.351   

       sqft_living15  sqft_lot15  
0               2060        7316  
1               2590       21891  
2               2250       20023  
3               1290       10320  
4               1480        3504  
...              ...         ...  
21431           1190        4168  
21432           1930        5150  
21433           1510        4538  
21434           1440        1188  
21435           1000        8100  

[21435 rows x 21 columns]

In [221]:
df.loc[df['bathrooms']== 8]

id       date      price  bedrooms  bathrooms  sqft_living  \
2670   1225069038 2014-05-05  2280000.0         7        8.0        13540   
14902  6762700020 2014-10-13  7700000.0         6        8.0        12050   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
2670     307752     3.0           0     4  ...     12        9410   
14902     27600     2.5           0     3  ...     13        8570   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
2670            4130      1999             0    98053  47.6675 -121.986   
14902           3480      1910          1987    98102  47.6298 -122.323   

       sqft_living15  sqft_lot15  
2670            4850      217800  
14902           3940        8800  

[2 rows x 21 columns]

Binning Prices

In [167]:
adf=df.copy()
bins = [0, 100000, 200000, 300000, 400000, 500000, 600000, 700000, 800000]
labels = ['0-100k', '100k-200k', '200k-300k', '300k-400k', '400k-500k', '500k-600k', '600k-700k', '700k-800k']

adf['bin_price'] = pd.cut(adf['price'], bins=bins, labels=labels, right=False)
adf



id       date     price  bedrooms  bathrooms  sqft_living  \
0         1000102 2015-04-22  300000.0         6       3.00         2400   
1         1200019 2014-05-08  647500.0         4       1.75         2060   
2         1200021 2014-08-11  400000.0         3       1.00         1460   
3         2800031 2015-04-01  235000.0         3       1.00         1430   
4         3600057 2015-03-19  402500.0         4       2.00         1650   
...           ...        ...       ...       ...        ...          ...   
21431  9842300095 2014-07-25  365000.0         5       2.00         1600   
21432  9842300485 2015-03-11  380000.0         2       1.00         1040   
21433  9842300540 2014-06-24  339000.0         3       1.00         1100   
21434  9895000040 2014-07-03  399900.0         2       1.75         1410   
21435  9900000190 2014-10-30  268950.0         3       1.00         1320   

       sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
0          9373     2.0           0     0  ...        2400              0   
1         26036     1.0           0     0  ...        1160            900   
2         43000     1.0           0     0  ...        1460              0   
3          7599     1.5           0     0  ...        1010            420   
4          3504     1.0           0     0  ...         760            890   
...         ...     ...         ...   ...  ...         ...            ...   
21431      4168     1.5           0     0  ...        1600              0   
21432      7372     1.0           0     0  ...         840            200   
21433      4128     1.0           0     0  ...         720            380   
21434      1005     1.5           0     0  ...         900            510   
21435      8100     1.0           0     0  ...         880            440   

       yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0          1991             0    98002  47.3262 -122.214           2060   
1          1947             0    98166  47.4444 -122.351           2590   
2          1952             0    98166  47.4434 -122.347           2250   
3          1930             0    98168  47.4783 -122.265           1290   
4          1951          2013    98144  47.5803 -122.294           1480   
...         ...           ...      ...      ...      ...            ...   
21431      1927             0    98126  47.5297 -122.381           1190   
21432      1939             0    98126  47.5285 -122.378           1930   
21433      1942             0    98126  47.5296 -122.379           1510   
21434      2011             0    98027  47.5446 -122.018           1440   
21435      1943             0    98166  47.4697 -122.351           1000   

       sqft_lot15  bin_price  
0            7316  300k-400k  
1           21891  600k-700k  
2           20023  400k-500k  
3           10320  200k-300k  
4            3504  400k-500k  
...           ...        ...  
21431        4168  300k-400k  
21432        5150  300k-400k  
21433        4538  300k-400k  
21434        1188  300k-400k  
21435        8100  200k-300k  

[21436 rows x 22 columns]

In [236]:
perc=adf.groupby('bin_price')['id'].count().to_frame().reset_index()
perc['percentage'] = perc['id'].apply(lambda x: 100 * x / 18445)
perc

bin_price    id  percentage
0     0-100k    20    0.108430
1  100k-200k   712    3.860125
2  200k-300k  3607   19.555435
3  300k-400k  4225   22.905937
4  400k-500k  3695   20.032529
5  500k-600k  2829   15.337490
6  600k-700k  1951   10.577392
7  700k-800k  1406    7.622662

Around 45 percent of the houses fall between the 300k to 500k bracket

Grade

In [242]:
grd=adf.copy()

In [259]:
grd['cat_grade'] = grd['grade'].apply(lambda x: 'very_high' if x > 8 else 'high' if 4 <= x <= 8 else 'low' if x < 4 else 'unknown')
grd[['cat_grade','bin_price','sqft_living','sqft_living15']].groupby(['bin_price','cat_grade']).mean().round(1).reset_index()
#sqft_living15 - The square footage of interior housing living space for the nearest 15 neighbors

bin_price  cat_grade  sqft_living  sqft_living15
0      0-100k       high        782.1         1274.8
1      0-100k        low        670.0         1160.0
2      0-100k  very_high          NaN            NaN
3   100k-200k       high       1118.0         1331.7
4   100k-200k        low        290.0         1620.0
5   100k-200k  very_high          NaN            NaN
6   200k-300k       high       1478.7         1559.3
7   200k-300k        low        560.0         1165.0
8   200k-300k  very_high       2035.0         2007.0
9   300k-400k       high       1714.5         1726.0
10  300k-400k        low          NaN            NaN
11  300k-400k  very_high       2463.3         2448.7
12  400k-500k       high       1801.7         1782.1
13  400k-500k        low          NaN            NaN
14  400k-500k  very_high       2755.6         2553.1
15  500k-600k       high       1989.2         1917.4
16  500k-600k        low          NaN            NaN
17  500k-600k  very_high       2703.1         2549.1
18  600k-700k       high       2155.4         2028.2
19  600k-700k        low          NaN            NaN
20  600k-700k  very_high       2753.4         2572.1
21  700k-800k       high       2294.0         2054.6
22  700k-800k        low          NaN            NaN
23  700k-800k  very_high       3004.7         2825.1

All of the houses starting from 200K are either high or very high grade
Low grade houses are on average very small in terms of living area and compared to the houses in the same neighborhood 
they are on average much much smaller.
The higher we go on the grading scale, the closer the sqft_living of the house matches that of the neighboring houses

Waterfront and View

In [181]:
adf[['bin_price', 'waterfront', 'view', 'grade']].groupby('bin_price').agg(['max', 'min', 'mean', 'median']).reset_index()

bin_price waterfront                      view                      grade  \
                    max min      mean median  max min      mean median   max   
0     0-100k          0   0  0.000000    0.0    0   0  0.000000    0.0     7   
1  100k-200k          0   0  0.000000    0.0    3   0  0.015449    0.0     8   
2  200k-300k          1   0  0.000554    0.0    4   0  0.040754    0.0     9   
3  300k-400k          1   0  0.001420    0.0    4   0  0.078817    0.0    10   
4  400k-500k          1   0  0.001353    0.0    4   0  0.122327    0.0    11   
5  500k-600k          1   0  0.002828    0.0    4   0  0.188052    0.0    11   
6  600k-700k          1   0  0.006663    0.0    4   0  0.254741    0.0    11   
7  700k-800k          1   0  0.006401    0.0    4   0  0.375533    0.0    11   

                        
  min      mean median  
0   3  5.200000    5.0  
1   1  6.307584    6.0  
2   3  6.838647    7.0  
3   4  7.216805    7.0  
4   4  7.421110    7.0  
5   5  7.709084    8.0  
6   5  8.030753    8.0  
7   5  8.393314    8.0

In [182]:
adf[['bin_price', 'waterfront', 'view', 'grade']].groupby(['waterfront','bin_price']).mean().round(1).reset_index()

waterfront  bin_price  view  grade
0            0     0-100k   0.0    5.2
1            0  100k-200k   0.0    6.3
2            0  200k-300k   0.0    6.8
3            0  300k-400k   0.1    7.2
4            0  400k-500k   0.1    7.4
5            0  500k-600k   0.2    7.7
6            0  600k-700k   0.2    8.0
7            0  700k-800k   0.4    8.4
8            1     0-100k   NaN    NaN
9            1  100k-200k   NaN    NaN
10           1  200k-300k   3.5    5.0
11           1  300k-400k   3.3    6.7
12           1  400k-500k   3.8    6.8
13           1  500k-600k   3.8    6.9
14           1  600k-700k   3.5    7.2
15           1  700k-800k   3.8    7.9

This shows us that the grade and waterfront aren't very strongly correlated, They remain pretty consistent for both waterfront and not waterfront, price seems to have more of an effect on grade.
Waterfront is more closely related to the view. But nither have much of an effect on the price in our dataset


YEAR BUILT



In [186]:
ydf=df.copy()
ydf['flag_yr_built']=adf.apply(lambda x:1 if x['yr_built']>1995 else 0, axis=1)
adf.groupby('flag_yr_built').agg({'price' : ['mean', 'max', 'min'],
                             'sqft_living' : ['mean', 'max', 'min'],
                             'sqft_lot' : ['mean', 'max', 'min'],
                             'condition' : ['median', 'max', 'min'],
                             'grade' : ['median', 'max', 'min'],
                             'id' : ['count']}).round(1)

price                      sqft_living             sqft_lot  \
                   mean        max       min        mean    max  min     mean   
flag_yr_built                                                                   
0              514283.5  7700000.0   75000.0      1939.1  12050  290  15598.8   
1              619082.5  6890000.0  155000.0      2487.1  13540  384  13831.7   

                            condition          grade             id  
                   max  min    median max min median max min  count  
flag_yr_built                                                        
0              1651359  520       3.0   5   1    7.0  13   1  15818  
1              1024068  572       3.0   5   3    8.0  13   4   5618

In [188]:
ydf.groupby('flag_yr_built').agg({'price' : ['mean', 'max', 'min'],
                             'bathrooms' : ['mean', 'max', 'min'],
                             'bedrooms' : ['mean', 'max', 'min'],
                             'floors' : ['median', 'max', 'min'],
                             'id' : ['count']}).round(1)

price                      bathrooms           bedrooms      \
                   mean        max       min      mean  max  min     mean max   
flag_yr_built                                                                   
0              514283.5  7700000.0   75000.0       1.9  8.0  0.0      3.3  33   
1              619082.5  6890000.0  155000.0       2.7  8.0  0.0      3.5  10   

                  floors               id  
              min median  max  min  count  
flag_yr_built                              
0               0    1.0  3.5  1.0  15818  
1               0    2.0  3.5  1.0   5618

Newer houses built have a higher grade, They are built much larger.
while the average number of bedrooms doesn't vary too much compared to houses built prior to 1995, It seems to be that new houses 
have overall more bathrooms built in, and the units generally are multi storeyed.

In [214]:
zip=ydf[['zipcode', 'price', 'flag_yr_built','id']].groupby('zipcode').agg({'id' : ['count']}).round(1).reset_index()
zip.loc[ydf['flag_yr_built']==1]

zipcode    id
           count
7    98008   283
13   98023   492
15   98027   411
17   98029   320
18   98030   253
19   98031   272
20   98032   123
28   98052   571
29   98053   402
31   98056   404
32   98058   451
35   98070   117
58   98136   263
62   98155   442
68   98198   275

Cities 98052, 98023, 98058 have the top 3 highest number of new housings built. Indicating more demand in housing.

In [177]:
df.corr()

id      date     price  bedrooms  bathrooms  sqft_living  \
id             1.000000  0.006322 -0.018471  0.000831   0.004344    -0.013467   
date           0.006322  1.000000 -0.006152 -0.017513  -0.035919    -0.035781   
price         -0.018471 -0.006152  1.000000  0.309187   0.524444     0.701427   
bedrooms       0.000831 -0.017513  0.309187  1.000000   0.516772     0.577512   
bathrooms      0.004344 -0.035919  0.524444  0.516772   1.000000     0.754422   
sqft_living   -0.013467 -0.035781  0.701427  0.577512   0.754422     1.000000   
sqft_lot      -0.132846  0.005862  0.088588  0.031719   0.087143     0.171957   
floors         0.018057 -0.023929  0.254959  0.175207   0.499863     0.352867   
waterfront    -0.002746  0.000945  0.266924 -0.006685   0.063734     0.103902   
view           0.010929 -0.001966  0.396659  0.079542   0.186684     0.283753   
condition     -0.024279 -0.051948  0.034555  0.027896  -0.127839    -0.060973   
grade          0.006416 -0.041525  0.666353  0.358396   0.664727     0.762402   
sqft_above    -0.011836 -0.029396  0.604629  0.478441   0.684994     0.876683   
sqft_basement -0.005795 -0.019238  0.323997  0.303035   0.283539     0.434419   
yr_built       0.020933 -0.001744  0.051051  0.153745   0.505092     0.316544   
yr_renovated  -0.017178 -0.025067  0.126722  0.018435   0.051101     0.055084   
zipcode       -0.007948  0.002811 -0.050940 -0.153621  -0.202958    -0.198621   
lat           -0.002859 -0.033164  0.306666 -0.010228   0.023444     0.051118   
long           0.019466 -0.008060  0.019366  0.129836   0.221935     0.239214   
sqft_living15 -0.004084 -0.032493  0.583937  0.393161   0.568194     0.756205   
sqft_lot15    -0.139670  0.002181  0.081684  0.029117   0.086627     0.182775   

               sqft_lot    floors  waterfront      view  ...     grade  \
id            -0.132846  0.018057   -0.002746  0.010929  ...  0.006416   
date           0.005862 -0.023929    0.000945 -0.001966  ... -0.041525   
price          0.088588  0.254959    0.266924  0.396659  ...  0.666353   
bedrooms       0.031719  0.175207   -0.006685  0.079542  ...  0.358396   
bathrooms      0.087143  0.499863    0.063734  0.186684  ...  0.664727   
sqft_living    0.171957  0.352867    0.103902  0.283753  ...  0.762402   
sqft_lot       1.000000 -0.005950    0.021571  0.075012  ...  0.112541   
floors        -0.005950  1.000000    0.023473  0.028211  ...  0.457475   
waterfront     0.021571  0.023473    1.000000  0.403025  ...  0.082841   
view           0.075012  0.028211    0.403025  1.000000  ...  0.249890   
condition     -0.009298 -0.266543    0.016607  0.045051  ... -0.148248   
grade          0.112541  0.457475    0.082841  0.249890  ...  1.000000   
sqft_above     0.182581  0.522648    0.072021  0.166930  ...  0.756068   
sqft_basement  0.015107 -0.246066    0.080838  0.276482  ...  0.167032   
yr_built       0.052447  0.489062   -0.026537 -0.055153  ...  0.445329   
yr_renovated   0.007636  0.005626    0.093025  0.103633  ...  0.014357   
zipcode       -0.129411 -0.058108    0.030558  0.087096  ... -0.182675   
lat           -0.085992  0.049279   -0.014395  0.005671  ...  0.113332   
long           0.229555  0.124285   -0.042188 -0.079720  ...  0.196572   
sqft_living15  0.143650  0.278854    0.086558  0.279077  ...  0.712507   
sqft_lot15     0.718097 -0.011917    0.030694  0.072867  ...  0.118313   

               sqft_above  sqft_basement  yr_built  yr_renovated   zipcode  \
id              -0.011836      -0.005795  0.020933     -0.017178 -0.007948   
date            -0.029396      -0.019238 -0.001744     -0.025067  0.002811   
price            0.604629       0.323997  0.051051      0.126722 -0.050940   
bedrooms         0.478441       0.303035  0.153745      0.018435 -0.153621   
bathrooms        0.684994       0.283539  0.505092      0.051101 -0.202958   
sqft_living      0.876683       0.434419  0.316544      0.055084 -0.198621   
sqft_lot         0.182581       0.015107  0.052447      0.00763